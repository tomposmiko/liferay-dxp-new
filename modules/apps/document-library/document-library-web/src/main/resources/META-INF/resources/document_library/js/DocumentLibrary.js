/**
 * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
 *
 * This library is free software; you can redistribute it and/or modify it under
 * the terms of the GNU Lesser General Public License as published by the Free
 * Software Foundation; either version 2.1 of the License, or (at your option)
 * any later version.
 *
 * This library is distributed in the hope that it will be useful, but WITHOUT
 * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
 * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
 * details.
 */

import {buildFragment} from 'frontend-js-web';

export default function DocumentLibrary({
	editEntryUrl,
	namespace,
	searchContainerId,
	selectFolderURL,
}) {
	let searchContainer;

	const form = document[`${namespace}fm2`];

	Liferay.componentReady(`${namespace}${searchContainerId}`).then(
		(searchContainerComponent) => {
			searchContainer = searchContainerComponent;

			searchContainer.registerAction('move-to-folder', _moveToFolder);
			searchContainer.registerAction('move-to-trash', _moveToTrash);
			searchContainer.on('rowToggled', _handleSearchContainerRowToggled);
		}
	);

	function _handleSearchContainerRowToggled() {
		const bulkSelection =
			searchContainer.select &&
			searchContainer.select.get('bulkSelection');

		form.elements[`${namespace}selectAll`].value = bulkSelection;
	}

	function _moveCurrentSelection(newFolderId) {
		form.action = editEntryUrl;
		form.method = 'POST';
		form.enctype = 'multipart/form-data';

		form.elements[`${namespace}cmd`].value = 'move';
		form.elements[`${namespace}newFolderId`].value = newFolderId;

		submitForm(form, editEntryUrl, false);
	}

	function _moveSingleElement(newFolderId, parameterName, parameterValue) {
		const redirectUrl = form.elements[`${namespace}redirect`].value;

		const newForm = document.createElement('div');

		newForm.appendChild(
			buildFragment(
				`<form action="${editEntryUrl}" class="hide" method="POST"><input name="${namespace}cmd" value="move"/>
					<input name="${namespace}newFolderId" value="${newFolderId}"/>
					<input name="${namespace}${parameterName}" value="${parameterValue}"/>
					<input name="${namespace}redirect" value="${redirectUrl}"/>
				</form>`
			)
		);

		const formNode = newForm.firstElementChild;

		form.append(formNode);

		submitForm(formNode, editEntryUrl, false);
	}

	function _moveToFolder(object) {
		const dropTarget = object.targetItem;

		const selectedItems = object.selectedItems;

		const folderId = dropTarget.attr('data-folder-id');

		if (folderId) {
			if (
				!searchContainer.select ||
				selectedItems.indexOf(dropTarget.one('input[type=checkbox]'))
			) {
				_moveCurrentSelection(folderId);
			}
		}
	}

	function _moveToTrash() {
		const action = 'move_to_trash';

		if (form.elements[`${namespace}javax-portlet-action`]) {
			form.elements[`${namespace}javax-portlet-action`].value = action;
		}
		else {
			form.elements[`${namespace}cmd`].value = action;
		}

		submitForm(form, editEntryUrl, false);
	}

	window[`${namespace}move`] = function (
		selectedItems,
		parameterName,
		parameterValue
	) {
		const dialogTitle =
			selectedItems === 1
				? Liferay.Language.get('select-destination-folder-for-x-item')
				: Liferay.Language.get('select-destination-folder-for-x-items');

		Liferay.Util.openSelectionModal({
			height: '480px',
			id: namespace,
			onSelect: (selectedItem) => {
				if (parameterName && parameterValue) {
					_moveSingleElement(
						selectedItem.folderid,
						parameterName,
						parameterValue
					);
				}
				else {
					_moveCurrentSelection(selectedItem.folderid);
				}
			},
			selectEventName: `${namespace}selectFolder`,
			size: 'lg',
			title: Liferay.Util.sub(dialogTitle, [selectedItems]),
			url: selectFolderURL,
		});
	};
}
