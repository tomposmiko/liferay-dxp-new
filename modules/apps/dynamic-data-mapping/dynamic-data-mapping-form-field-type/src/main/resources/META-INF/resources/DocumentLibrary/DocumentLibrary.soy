{namespace DocumentLibrary}

{template .render}
	{@param name: string}
	{@param spritemap: string}
	{@param? _handleClearButtonClicked: any}
	{@param? _handleSelectButtonClicked: any}
	{@param? displayErrors: bool}
	{@param? errorMessage: string}
	{@param? fileEntryTitle: string}
	{@param? fileEntryURL: string}
	{@param? id: string}
	{@param? label: string}
	{@param? placeholder: string}
	{@param? readOnly: bool}
	{@param? repeatable: bool}
	{@param? required: bool}
	{@param? showLabel: bool}
	{@param? tip: string}
	{@param? tooltip: string}
	{@param? valid: bool}
	{@param? value: ?}
	{@param? visible: bool}

	{call FieldBase.render}
		{param contentRenderer kind="html"}
			{call .content}
				{param _handleClearButtonClicked: $_handleClearButtonClicked /}
				{param _handleSelectButtonClicked: $_handleSelectButtonClicked /}
				{param fileEntryTitle: $fileEntryTitle /}
				{param fileEntryURL: $fileEntryURL /}
				{param id: $id /}
				{param name: $name /}
				{param placeholder: $placeholder /}
				{param readOnly: $readOnly /}
				{param spritemap: $spritemap /}
				{param value: $value /}
			{/call}
		{/param}
		{param displayErrors: $displayErrors /}
		{param errorMessage: $errorMessage /}
		{param id: $id /}
		{param label: $label /}
		{param name: $name /}
		{param readOnly: $readOnly /}
		{param repeatable: $repeatable /}
		{param required: $required /}
		{param showLabel: $showLabel /}
		{param spritemap: $spritemap /}
		{param tip: $tip /}
		{param tooltip: $tooltip /}
		{param valid: $valid /}
		{param visible: $visible /}
	{/call}
{/template}

{template .content}
	{@param name: string}
	{@param spritemap: string}
	{@param? _handleClearButtonClicked: any}
	{@param? _handleSelectButtonClicked: any}
	{@param? fileEntryTitle: string}
	{@param? fileEntryURL: string}
	{@param? id: string}
	{@param? placeholder: string}
	{@param? readOnly: bool}
	{@param? value: ?}

	{let $attributes kind="attributes"}
		{if $id}
			id="{$id}"
		{/if}

		name="{$name}"

		type="hidden"

		{if $placeholder}
			placeholder="{$placeholder}"
		{/if}

		{if $value}
			value="{$value}"
		{else}
			value=""
		{/if}
	{/let}

	<div class="liferay-ddm-form-field-document-library">
		{if $fileEntryURL and $readOnly}
			{call .card_item}
				{param fileEntryTitle: $fileEntryTitle /}
				{param fileEntryURL: $fileEntryURL /}
				{param spritemap: $spritemap /}
			{/call}
		{else}
			<div class="input-group">
				<div class="input-group-item input-group-prepend">
					<input
						class="field form-control"
						disabled
						id="{$name}inputFile"
						type="text"

						{if $fileEntryTitle}
							value="{$fileEntryTitle}"
						{else}
							value=""
						{/if}
					/>
				</div>

				{call .button_group}
					{param _handleClearButtonClicked: $_handleClearButtonClicked /}
					{param _handleSelectButtonClicked: $_handleSelectButtonClicked /}
					{param fileEntryURL: $fileEntryURL /}
					{param readOnly: $readOnly /}
					{param spritemap: $spritemap /}
				{/call}
			</div>
		{/if}

		<input {$attributes} />
	</div>
{/template}

/**
 * Prints the button group to manipulate files.
 */
{template .button_group}
	{@param? _handleClearButtonClicked: any}
	{@param? _handleSelectButtonClicked: any}
	{@param? readOnly: bool}
	{@param? fileEntryURL: string}
	{@param spritemap: string}

	<div class="input-group-append input-group-item input-group-item-shrink">
		<button
			class="btn btn-secondary select-button"
			data-onclick="{$_handleSelectButtonClicked}"

			{if $readOnly}
				disabled
			{/if}

			type="button"
		>
			<span class="lfr-btn-label">{msg desc=""}select{/msg}</span>
		</button>
	</div>

	{if $fileEntryURL}
		<div class="input-group-append input-group-item input-group-item-shrink">
			<button
				class="btn btn-secondary clear-button"
				data-onclick="{$_handleClearButtonClicked}"
				type="button"
			>
				<svg class="lexicon-icon">
					<use xlink:href="{$spritemap}#times" />
				</svg>
			</button>
		</div>
	{/if}
{/template}

/**
 * Prints the card to allow file to be downloaded.
 */
{template .card_item}
	{@param spritemap: string}
	{@param? fileEntryTitle: string}
	{@param? fileEntryURL: string}

	<div class="card card-horizontal">
		<div class="card-row card-row-padded">
			<div class="card-col-content card-col-gutters">
				<h4 class="truncate-text" title="{$fileEntryTitle}">{$fileEntryTitle}</h4>
			</div>

			<div class="card-col-field">
				<a download="{$fileEntryTitle}" href="{$fileEntryURL}">
					<svg class="lexicon-icon">
						<use xlink:href="{$spritemap}#download" />
					</svg>
				</a>
			</div>
		</div>
	</div>
{/template}