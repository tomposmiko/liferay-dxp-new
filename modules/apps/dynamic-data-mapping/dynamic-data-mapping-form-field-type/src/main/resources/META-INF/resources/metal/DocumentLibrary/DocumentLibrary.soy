{namespace DocumentLibrary}

{template .render}
	{@param name: string}
	{@param strings: [select: string]}
	{@param? id: string}
	{@param? label: string}
	{@param? placeholder: string}
	{@param? readOnly: bool}
	{@param? repeatable: bool}
	{@param? required: bool}
	{@param? showLabel: bool}
	{@param? spritemap: string}
	{@param? tip: string}
	{@param? tooltip: string}
	{@param? value: ?}

	{call FieldBase.render}
		{param contentRenderer kind="html"}
			{call .content}
				{param id: $id /}
				{param name: $name /}
				{param placeholder: $placeholder /}
				{param readOnly: $readOnly /}
				{param spritemap: $spritemap /}
				{param strings: $strings /}
				{param tip: $tip /}
				{param value: $value /}
			{/call}
		{/param}
		{param id: $id /}
		{param label: $label /}
		{param repeatable: $repeatable /}
		{param required: $required /}
		{param showLabel: $showLabel /}
		{param spritemap: $spritemap /}
		{param tip: $tip /}
		{param tooltip: $tooltip /}
	{/call}
{/template}

{template .content}
	{@param name: string}
	{@param strings: [select: string]}
	{@param? id: string}
	{@param? placeholder: string}
	{@param? readOnly: bool}
	{@param? showLabel: bool}
	{@param? tip: string}
	{@param? value: ?}

	{let $attributes kind="attributes"}
		{if $id}
			id="{$id}"
		{/if}

		name="{$name}"

		type="hidden"

		{if $placeholder}
			placeholder="{$placeholder}"
		{/if}

		{if $value}
			value="{$value}"
		{else}
			value=""
		{/if}
	{/let}

	<div class="liferay-ddm-form-field-document-library">
		<div class="input-group">
			<div class="input-group-item input-group-prepend">
				<input
					class="field form-control"
					disabled
					id="{$name}inputFile"
					type="text"
				/>
			</div>

			{call .button_group}
				{param readOnly: $readOnly /}
				{param strings: $strings /}
			{/call}
		</div>

		<input {$attributes} />
	</div>

	{if $showLabel}
		{if $tip}
			<span class="form-text">{$tip}</span>
		{/if}
	{/if}
{/template}

/**
 * Prints the button group to manipulate files.
 */
{template .button_group}
	{@param strings: [select: string]}
	{@param? readOnly: bool}

	<div class="input-group-append input-group-item input-group-item-shrink">
		<button class="btn btn-secondary select-button" {if $readOnly}disabled{/if} type="button">
			<span class="lfr-btn-label">{$strings.select}</span>
		</button>
	</div>
{/template}