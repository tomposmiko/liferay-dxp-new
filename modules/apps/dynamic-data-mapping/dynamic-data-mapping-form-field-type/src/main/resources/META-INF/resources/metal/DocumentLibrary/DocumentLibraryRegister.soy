{namespace DocumentLibraryRegister}

{deltemplate PageRenderer.RegisterFieldType variant="'document_library'"}
	{call DocumentLibrary.render data="all" /}
{/deltemplate}