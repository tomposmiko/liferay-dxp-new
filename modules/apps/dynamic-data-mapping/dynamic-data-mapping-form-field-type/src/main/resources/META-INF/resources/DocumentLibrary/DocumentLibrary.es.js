/**
 * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
 *
 * This library is free software; you can redistribute it and/or modify it under
 * the terms of the GNU Lesser General Public License as published by the Free
 * Software Foundation; either version 2.1 of the License, or (at your option)
 * any later version.
 *
 * This library is distributed in the hope that it will be useful, but WITHOUT
 * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
 * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
 * details.
 */

import '../FieldBase/FieldBase.es';

import './DocumentLibraryRegister.soy.js';

import {createActionURL, createPortletURL} from 'frontend-js-web';
import Component from 'metal-component';
import Soy from 'metal-soy';
import {Config} from 'metal-state';

import templates from './DocumentLibrary.soy.js';

class DocumentLibrary extends Component {
	created() {
		AUI().use('liferay-item-selector-dialog', A => {
			this.A = A;
		});
	}

	prepareStateForRender(state) {
		let {fileEntryTitle, fileEntryURL, value} = state;

		if (value) {
			if (typeof value === 'object') {
				fileEntryTitle = value.title;
				fileEntryURL = value.url;

				value = JSON.stringify(value);
			} else if (typeof value === 'string') {
				const object = JSON.parse(value);

				fileEntryTitle = object.title;
				fileEntryURL = object.url;
			}
		}

		return {
			...state,
			fileEntryTitle,
			fileEntryURL,
			value
		};
	}

	getDocumentLibrarySelectorURL() {
		const {itemSelectorAuthToken, portletNamespace} = this;

		const criterionJSON = {
			desiredItemSelectorReturnTypes:
				'com.liferay.item.selector.criteria.FileEntryItemSelectorReturnType,com.liferay.item.selector.criteria.FileEntryItemSelectorReturnType'
		};

		const uploadCriterionJSON = {
			URL: this.getUploadURL(),
			desiredItemSelectorReturnTypes:
				'com.liferay.item.selector.criteria.FileEntryItemSelectorReturnType,com.liferay.item.selector.criteria.FileEntryItemSelectorReturnType'
		};

		const documentLibrarySelectorParameters = {
			'0_json': JSON.stringify(criterionJSON),
			'1_json': JSON.stringify(criterionJSON),
			'2_json': JSON.stringify(uploadCriterionJSON),
			criteria:
				'com.liferay.item.selector.criteria.file.criterion.FileItemSelectorCriterion',
			doAsGroupId: themeDisplay.getScopeGroupId(),
			itemSelectedEventName: `${portletNamespace}selectDocumentLibrary`,
			p_p_auth: itemSelectorAuthToken,
			p_p_id: Liferay.PortletKeys.ITEM_SELECTOR,
			p_p_mode: 'view',
			p_p_state: 'pop_up',
			refererGroupId: themeDisplay.getScopeGroupId()
		};

		const documentLibrarySelectorURL = createPortletURL(
			themeDisplay.getLayoutRelativeControlPanelURL(),
			documentLibrarySelectorParameters
		);

		return documentLibrarySelectorURL.toString();
	}

	getUploadURL() {
		const uploadParameters = {
			cmd: 'add_temp',
			'javax.portlet.action': '/document_library/upload_file_entry',
			p_auth: Liferay.authToken,
			p_p_id: Liferay.PortletKeys.DOCUMENT_LIBRARY
		};

		const uploadURL = createActionURL(
			themeDisplay.getLayoutRelativeURL(),
			uploadParameters
		);

		return uploadURL.toString();
	}

	_handleClearButtonClicked() {
		this.setState({
			value: ''
		});
	}

	_handleFieldChanged(event) {
		var selectedItem = event.newVal;

		if (selectedItem) {
			const {value} = selectedItem;

			this.setState(
				{
					value
				},
				() => {
					this.emit('fieldEdited', {
						fieldInstance: this,
						originalEvent: event,
						value
					});
				}
			);
		}
	}

	_handleSelectButtonClicked() {
		var {A, portletNamespace} = this;

		var itemSelectorDialog = new A.LiferayItemSelectorDialog({
			eventName: `${portletNamespace}selectDocumentLibrary`,
			on: {
				selectedItemChange: this._handleFieldChanged.bind(this),
				visibleChange: this._handleVisibleChange.bind(this)
			},
			url: this.getDocumentLibrarySelectorURL()
		});

		itemSelectorDialog.open();
	}

	_handleVisibleChange(event) {
		if (event.newVal) {
			this.emit('fieldFocused', {
				fieldInstance: this,
				originalEvent: event
			});
		} else {
			this.emit('fieldBlurred', {
				fieldInstance: this,
				originalEvent: event
			});
		}
	}
}

DocumentLibrary.STATE = {
	/**
	 * @default 'string'
	 * @memberof Text
	 * @type {?(string|undefined)}
	 */

	dataType: Config.string().value('date'),

	/**
	 * @default false
	 * @memberof DocumentLibrary
	 * @type {?bool}
	 */

	evaluable: Config.bool().value(false),

	/**
	 * @default undefined
	 * @memberof DocumentLibrary
	 * @type {?(string|undefined)}
	 */

	fieldName: Config.string(),

	/**
	 * @default undefined
	 * @memberof DocumentLibrary
	 * @type {?(string|undefined)}
	 */

	id: Config.string(),

	/**
	 * @default undefined
	 * @memberof DocumentLibrary
	 * @type {?(string|undefined)}
	 */

	label: Config.string(),

	/**
	 * @default undefined
	 * @memberof DocumentLibrary
	 * @type {?(string|undefined)}
	 */

	name: Config.string().required(),

	/**
	 * @default undefined
	 * @memberof DocumentLibrary
	 * @type {?(string|undefined)}
	 */

	placeholder: Config.string(),

	/**
	 * @default false
	 * @memberof DocumentLibrary
	 * @type {?bool}
	 */

	readOnly: Config.bool().value(false),

	/**
	 * @default undefined
	 * @memberof FieldBase
	 * @type {?(bool|undefined)}
	 */

	repeatable: Config.bool(),

	/**
	 * @default false
	 * @memberof DocumentLibrary
	 * @type {?(bool|undefined)}
	 */

	required: Config.bool().value(false),

	/**
	 * @default true
	 * @memberof DocumentLibrary
	 * @type {?(bool|undefined)}
	 */

	showLabel: Config.bool().value(true),

	/**
	 * @default undefined
	 * @memberof DocumentLibrary
	 * @type {?(string|undefined)}
	 */

	spritemap: Config.string(),

	/**
	 * @default undefined
	 * @memberof DocumentLibrary
	 * @type {?(string|undefined)}
	 */

	tip: Config.string(),

	/**
	 * @default undefined
	 * @memberof FieldBase
	 * @type {?(string|undefined)}
	 */

	tooltip: Config.string(),

	/**
	 * @default undefined
	 * @memberof Text
	 * @type {?(string|undefined)}
	 */

	type: Config.string().value('document_library'),

	/**
	 * @default undefined
	 * @memberof DocumentLibrary
	 * @type {?(string|undefined)}
	 */

	value: Config.oneOfType([Config.object(), Config.string()])
};

Soy.register(DocumentLibrary, templates);

export default DocumentLibrary;
