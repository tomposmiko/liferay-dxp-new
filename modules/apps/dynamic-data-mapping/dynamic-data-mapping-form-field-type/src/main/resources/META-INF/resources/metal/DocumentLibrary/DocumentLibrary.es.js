import './DocumentLibraryRegister.soy.js';
import 'dynamic-data-mapping-form-field-type/metal/FieldBase/index.es';
import Component from 'metal-component';
import Soy from 'metal-soy';
import templates from './DocumentLibrary.soy.js';
import {Config} from 'metal-state';

class DocumentLibrary extends Component {
	static STATE = {

		/**
		 * @default 'string'
		 * @instance
		 * @memberof Text
		 * @type {?(string|undefined)}
		 */

		dataType: Config.string().value('date'),

		/**
		 * @default undefined
		 * @instance
		 * @memberof DocumentLibrary
		 * @type {?(string|undefined)}
		 */

		fieldName: Config.string(),

		/**
		 * @default undefined
		 * @instance
		 * @memberof DocumentLibrary
		 * @type {?(string|undefined)}
		 */

		id: Config.string(),

		/**
		 * @default undefined
		 * @instance
		 * @memberof DocumentLibrary
		 * @type {?(string|undefined)}
		 */

		label: Config.string(),

		/**
		 * @default undefined
		 * @instance
		 * @memberof DocumentLibrary
		 * @type {?(string|undefined)}
		 */

		name: Config.string().required(),

		/**
		 * @default undefined
		 * @instance
		 * @memberof DocumentLibrary
		 * @type {?(string|undefined)}
		 */

		placeholder: Config.string(),

		/**
		 * @default false
		 * @instance
		 * @memberof DocumentLibrary
		 * @type {?bool}
		 */

		readOnly: Config.bool().value(false),

		/**
		 * @default false
		 * @instance
		 * @memberof DocumentLibrary
		 * @type {?(bool|undefined)}
		 */

		required: Config.bool().value(false),

		/**
		 * @default undefined
		 * @instance
		 * @memberof FieldBase
		 * @type {?(bool|undefined)}
		 */

		repeatable: Config.bool(),

		/**
		 * @default true
		 * @instance
		 * @memberof DocumentLibrary
		 * @type {?(bool|undefined)}
		 */

		showLabel: Config.bool().value(true),

		/**
		 * @default undefined
		 * @instance
		 * @memberof DocumentLibrary
		 * @type {?(string|undefined)}
		 */

		spritemap: Config.string(),

		strings: {
			value: {
				select: Liferay.Language.get('select')
			}
		},

		/**
		 * @default undefined
		 * @instance
		 * @memberof DocumentLibrary
		 * @type {?(string|undefined)}
		 */

		tip: Config.string(),

		/**
		 * @default undefined
		 * @instance
		 * @memberof FieldBase
		 * @type {?(string|undefined)}
		 */

		tooltip: Config.string(),

		/**
		 * @default undefined
		 * @instance
		 * @memberof Text
		 * @type {?(string|undefined)}
		 */

		type: Config.string().value('document_library'),

		/**
		 * @default undefined
		 * @instance
		 * @memberof DocumentLibrary
		 * @type {?(string|undefined)}
		 */

		value: Config.object()
	};
}

Soy.register(DocumentLibrary, templates);

export default DocumentLibrary;