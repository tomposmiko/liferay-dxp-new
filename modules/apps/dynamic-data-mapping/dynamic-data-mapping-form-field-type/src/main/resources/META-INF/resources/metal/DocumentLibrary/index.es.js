import DocumentLibrary from './DocumentLibrary.es';

export default DocumentLibrary;