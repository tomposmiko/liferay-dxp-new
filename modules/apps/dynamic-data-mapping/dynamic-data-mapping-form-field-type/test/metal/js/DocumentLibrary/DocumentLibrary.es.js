import DocumentLibrary from 'source/DocumentLibrary/DocumentLibrary.es';

let component;
const spritemap = 'icons.svg';

const defaultDocumentLibraryConfig = {
	name: 'textField',
	spritemap
};

describe(
	'Field DocumentLibrary',
	() => {
		afterEach(
			() => {
				if (component) {
					component.dispose();
				}
			}
		);

		it(
			'should not be readOnly',
			() => {
				component = new DocumentLibrary(
					{
						...defaultDocumentLibraryConfig,
						readOnly: false
					}
				);

				expect(component).toMatchSnapshot();
			}
		);

		it(
			'should have a helptext',
			() => {
				component = new DocumentLibrary(
					{
						...defaultDocumentLibraryConfig,
						tip: 'Type something'
					}
				);

				expect(component).toMatchSnapshot();
			}
		);

		it(
			'should have an id',
			() => {
				component = new DocumentLibrary(
					{
						...defaultDocumentLibraryConfig,
						id: 'ID'
					}
				);

				expect(component).toMatchSnapshot();
			}
		);

		it(
			'should have a label',
			() => {
				component = new DocumentLibrary(
					{
						...defaultDocumentLibraryConfig,
						label: 'label'
					}
				);

				expect(component).toMatchSnapshot();
			}
		);

		it(
			'should have a placeholder',
			() => {
				component = new DocumentLibrary(
					{
						...defaultDocumentLibraryConfig,
						placeholder: 'Placeholder'
					}
				);

				expect(component).toMatchSnapshot();
			}
		);

		it(
			'should not be required',
			() => {
				component = new DocumentLibrary(
					{
						...defaultDocumentLibraryConfig,
						required: false
					}
				);

				expect(component).toMatchSnapshot();
			}
		);

		it(
			'should render Label if showLabel is true',
			() => {
				component = new DocumentLibrary(
					{
						...defaultDocumentLibraryConfig,
						label: 'text',
						showLabel: true
					}
				);

				expect(component).toMatchSnapshot();
			}
		);

		it(
			'should have a spritemap',
			() => {
				component = new DocumentLibrary(defaultDocumentLibraryConfig);

				expect(component).toMatchSnapshot();
			}
		);

		it(
			'should have a value',
			() => {
				component = new DocumentLibrary(
					{
						...defaultDocumentLibraryConfig,
						value: {
							id: '123'
						}
					}
				);

				expect(component).toMatchSnapshot();
			}
		);
	}
);