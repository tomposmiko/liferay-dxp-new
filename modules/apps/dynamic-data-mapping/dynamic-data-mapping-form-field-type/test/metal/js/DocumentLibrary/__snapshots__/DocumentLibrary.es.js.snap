// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Field DocumentLibrary should have a helptext 1`] = `
<div class="form-group">
  <div class="liferay-ddm-form-field-document-library">
    <div class="input-group">
      <div class="input-group-item input-group-prepend">
        <input class="field form-control" disabled="" id="textFieldinputFile" type="text">
      </div>
      <div class="input-group-append input-group-item input-group-item-shrink">
        <button class="btn btn-secondary select-button" type="button">
          <span class="lfr-btn-label">select</span>
        </button>
      </div>
    </div>
    <input name="textField" type="hidden" value="">
  </div>
  <span class="form-text">Type something</span>
</div>
`;

exports[`Field DocumentLibrary should have a label 1`] = `
<div class="form-group">
  <label>label </label>
  <div class="liferay-ddm-form-field-document-library">
    <div class="input-group">
      <div class="input-group-item input-group-prepend">
        <input class="field form-control" disabled="" id="textFieldinputFile" type="text">
      </div>
      <div class="input-group-append input-group-item input-group-item-shrink">
        <button class="btn btn-secondary select-button" type="button">
          <span class="lfr-btn-label">select</span>
        </button>
      </div>
    </div>
    <input name="textField" type="hidden" value="">
  </div>
</div>
`;

exports[`Field DocumentLibrary should have a placeholder 1`] = `
<div class="form-group">
  <div class="liferay-ddm-form-field-document-library">
    <div class="input-group">
      <div class="input-group-item input-group-prepend">
        <input class="field form-control" disabled="" id="textFieldinputFile" type="text">
      </div>
      <div class="input-group-append input-group-item input-group-item-shrink">
        <button class="btn btn-secondary select-button" type="button">
          <span class="lfr-btn-label">select</span>
        </button>
      </div>
    </div>
    <input name="textField" type="hidden" placeholder="Placeholder" value="">
  </div>
</div>
`;

exports[`Field DocumentLibrary should have a spritemap 1`] = `
<div class="form-group">
  <div class="liferay-ddm-form-field-document-library">
    <div class="input-group">
      <div class="input-group-item input-group-prepend">
        <input class="field form-control" disabled="" id="textFieldinputFile" type="text">
      </div>
      <div class="input-group-append input-group-item input-group-item-shrink">
        <button class="btn btn-secondary select-button" type="button">
          <span class="lfr-btn-label">select</span>
        </button>
      </div>
    </div>
    <input name="textField" type="hidden" value="">
  </div>
</div>
`;

exports[`Field DocumentLibrary should have a value 1`] = `
<div class="form-group">
  <div class="liferay-ddm-form-field-document-library">
    <div class="input-group">
      <div class="input-group-item input-group-prepend">
        <input class="field form-control" disabled="" id="textFieldinputFile" type="text">
      </div>
      <div class="input-group-append input-group-item input-group-item-shrink">
        <button class="btn btn-secondary select-button" type="button">
          <span class="lfr-btn-label">select</span>
        </button>
      </div>
    </div>
    <input name="textField" type="hidden" value="[object Object]">
  </div>
</div>
`;

exports[`Field DocumentLibrary should have an id 1`] = `
<div class="form-group">
  <div class="liferay-ddm-form-field-document-library">
    <div class="input-group">
      <div class="input-group-item input-group-prepend">
        <input class="field form-control" disabled="" id="textFieldinputFile" type="text">
      </div>
      <div class="input-group-append input-group-item input-group-item-shrink">
        <button class="btn btn-secondary select-button" type="button">
          <span class="lfr-btn-label">select</span>
        </button>
      </div>
    </div>
    <input id="ID" name="textField" type="hidden" value="">
  </div>
</div>
`;

exports[`Field DocumentLibrary should not be readOnly 1`] = `
<div class="form-group">
  <div class="liferay-ddm-form-field-document-library">
    <div class="input-group">
      <div class="input-group-item input-group-prepend">
        <input class="field form-control" disabled="" id="textFieldinputFile" type="text">
      </div>
      <div class="input-group-append input-group-item input-group-item-shrink">
        <button class="btn btn-secondary select-button" type="button">
          <span class="lfr-btn-label">select</span>
        </button>
      </div>
    </div>
    <input name="textField" type="hidden" value="">
  </div>
</div>
`;

exports[`Field DocumentLibrary should not be required 1`] = `
<div class="form-group">
  <div class="liferay-ddm-form-field-document-library">
    <div class="input-group">
      <div class="input-group-item input-group-prepend">
        <input class="field form-control" disabled="" id="textFieldinputFile" type="text">
      </div>
      <div class="input-group-append input-group-item input-group-item-shrink">
        <button class="btn btn-secondary select-button" type="button">
          <span class="lfr-btn-label">select</span>
        </button>
      </div>
    </div>
    <input name="textField" type="hidden" value="">
  </div>
</div>
`;

exports[`Field DocumentLibrary should render Label if showLabel is true 1`] = `
<div class="form-group">
  <label>text </label>
  <div class="liferay-ddm-form-field-document-library">
    <div class="input-group">
      <div class="input-group-item input-group-prepend">
        <input class="field form-control" disabled="" id="textFieldinputFile" type="text">
      </div>
      <div class="input-group-append input-group-item input-group-item-shrink">
        <button class="btn btn-secondary select-button" type="button">
          <span class="lfr-btn-label">select</span>
        </button>
      </div>
    </div>
    <input name="textField" type="hidden" value="">
  </div>
</div>
`;
